pkg/ddevapp/craftcms.go

```diff
@@ -111,42 +86,32 @@ func craftCmsPostStartAction(app *DdevApp) error {
 		port = "5432"
 	}
 
-	// If they have older version of .env with DB_DRIVER, DB_SERVER etc, use those
-	if _, ok := envMap["DB_SERVER"]; ok {
-		// TODO: Remove, was never an official standard of Craft CMS.
-		envMap = map[string]string{
-			"DB_DRIVER":             driver,
-			"DB_SERVER":             "db",
-			"DB_PORT":               port,
-			"DB_DATABASE":           "db",
-			"DB_USER":               "db",
-			"DB_PASSWORD":           "db",
-			"MAILPIT_SMTP_HOSTNAME": "127.0.0.1",
-			"MAILPIT_SMTP_PORT":     "1025",
-			"PRIMARY_SITE_URL":      app.GetPrimaryURL(),
-		}
-	} else {
-		// Otherwise use the current CRAFT_DB_SERVER etc.
-		envMap = map[string]string{
-			"CRAFT_DB_DRIVER":       driver,
-			"CRAFT_DB_SERVER":       "db",
-			"CRAFT_DB_PORT":         port,
-			"CRAFT_DB_DATABASE":     "db",
-			"CRAFT_DB_USER":         "db",
-			"CRAFT_DB_PASSWORD":     "db",
-			"CRAFT_WEB_ROOT":        app.GetAbsDocroot(true),
-			"MAILPIT_SMTP_HOSTNAME": "127.0.0.1",
-			"MAILPIT_SMTP_PORT":     "1025",
-			"PRIMARY_SITE_URL":      app.GetPrimaryURL(),
-		}
+	newCraftEnvMap := map[string]string{
+		"CRAFT_DB_DRIVER":       driver,
+		"CRAFT_DB_SERVER":       "db",
+		"CRAFT_DB_PORT":         port,
+		"CRAFT_DB_DATABASE":     "db",
+		"CRAFT_DB_USER":         "db",
+		"CRAFT_DB_PASSWORD":     "db",
+		"CRAFT_WEB_ROOT":        app.GetAbsDocroot(true),
+		"MAILPIT_SMTP_HOSTNAME": "127.0.0.1",
+		"MAILPIT_SMTP_PORT":     "1025",
+		"PRIMARY_SITE_URL":      app.GetPrimaryURL(),
+	}
+
+	// Read existing env file
+	_, existingEnvText, err := ReadProjectEnvFile(envFilePath)
+	// If envFilePath doesn't exist, that's not really an error, continue
+	if !os.IsNotExist(err) {
```

Comment:
If I run:

```
ddev dotenv set .ddev/.env.web --craft-db-user=foobar && ddev config --auto
```

`CRAFT_DB_USER` is not updated anymore, that because of this missing condition:

```suggestion
	if err != nil && !os.IsNotExist(err) {
```